include/blas_quda.h

```diff
@@ -26,126 +26,458 @@ namespace quda {
     extern unsigned long long flops;
     extern unsigned long long bytes;
 
-    void zero(ColorSpinorField &a);
+    inline void zero(cvector_ref<ColorSpinorField> &x)
+    {
+      for (auto &xi : x) xi.get().zero();
```

Comment:
just wondering if for cases like that we could come up with a way to do something to get rid of the .get(). But maybe there just is no easy/nice/clean way around this here.

Reply:
If we use the array accessor, we can avoid the `get()` since I overloaded it return the underlying reference.

[b37ed336](https://github.com/lattice/quda/commit/b37ed336c69a82e851d2cfccd3ca5472c750f718)